festim/exports/txt_export.py

```diff
@@ -46,22 +45,61 @@ def when_is_next_time(self, current_time):
                 return time
         return None
 
-    def write(self, current_time, steady):
+    def is_it_first_time_to_export(self, current_time, nb_iteration):
+        if self.times is None:
+            return True if nb_iteration == 0 else False
+        else:
+            return True if np.isclose(self.times[0], current_time) else False
+
+    def write(self, current_time, nb_iteration, steady):
         # create a DG1 functionspace
         V_DG1 = f.FunctionSpace(self.function.function_space().mesh(), "DG", 1)
 
         solution = f.project(self.function, V_DG1)
         if self.is_it_time_to_export(current_time):
-            filename = "{}/{}_{}s.txt".format(self.folder, self.label, current_time)
             if steady:
                 filename = "{}/{}_steady.txt".format(self.folder, self.label)
+                header = "x,t=steady"
+            else:
+                filename = "{}/{}_transient.txt".format(self.folder, self.label)
+                header = "x,t={}s".format(current_time)
             x = f.interpolate(f.Expression("x[0]", degree=1), V_DG1)
             # if the directory doesn't exist
             # create it
             dirname = os.path.dirname(filename)
             if not os.path.exists(dirname):
                 os.makedirs(dirname, exist_ok=True)
-            np.savetxt(filename, np.transpose([x.vector()[:], solution.vector()[:]]))
+            # if steady or it is the first time to export
+            # write data
+            # else append new column to the existing file
+            if steady or self.is_it_first_time_to_export(current_time, nb_iteration):
+                np.savetxt(
+                    filename,
+                    np.transpose([x.vector()[:], solution.vector()[:]]),
+                    header=header,
+                    delimiter=",",
+                    comments="",
```

Comment:
not sure if this attribute is needed unless you had another use in mind for it?

Reply:
I think this is needed because otherwise the default value for coments is `"#"` and it adds this symbol to the header.
Correct @KulaginVladimir ?

Reply:
@RemDelaporteMathurin, right. I think `comments` should certainly be turned off when new columns are added to the output file